### PR TITLE
Add a --mode=local flag to kubernetes_e2e scenario

### PR DESCRIPTION
**Pull request by fejta**
`kubernetes_e2e.py --mode=docker # the default` runs the scenario via `docker run kubekins`
`kubernetes_e2e.py --mode=local` runs the scenario by starting a e2e-runner.sh subprocess.

Also:
* fix the binary name in bootstrap.py -h
* Delete unused cp /var/lib/jenkins gce keys (we always run on docker now in jenkins)
* Delete non bootstrap migration mode (everything uses bootstrap now)

### CODE REVIEW

**Comment by fejta** [comment]
For testing I'll edit the canary job on my local jobs/config.json to add a --mode=local line and then do this:

`$GOPATH/src/k8s.io/test-infra/jenkins/bootstrap.py --bare --json --job=ci-kubernetes-e2e-gce-canary --service-account ~/kubekins.json`

File: scenarios/kubernetes_e2e.py [hunk 1]
```diff
@@ -44,49 +45,169 @@ def check(*cmd):
...
+        if not os.path.isfile(os.path.join(parent, 'kubetest')):
+            check('go', 'install', 'k8s.io/test-infra/kubetest')
+            shutil.copy(
+                os.path.expandvars('${GOPATH}/bin/kubetest'),
+                os.path.join(parent, 'kubetest'))
+
+    def start(self):
+        """Runs e2e-runner.sh after setting env and installing prereqs."""
+        env = {}
+        env.update(self.env_files)
```
Comment (krzyzacy) [inline scenarios/kubernetes_e2e.py]: really?! this is like black magic...

Comment (fejta) [inline scenarios/kubernetes_e2e.py]: heh, `help(dict.update)`

Comment (krzyzacy) [inline scenarios/kubernetes_e2e.py]: Oh... it's parsed above.. nvm.. I was thinking python has a native env-file handler or something..

Comment (krzyzacy) [inline scenarios/kubernetes_e2e.py]: Like env.update('foo.env') and it will do the magic :-p maybe in the future :-p

Comment (fejta) [inline scenarios/kubernetes_e2e.py]: Well I'm really happy with your strict enforcement of the .env file format, makes it easy to parse here.

File: scenarios/kubernetes_e2e.py [hunk 2]
```diff
@@ -44,49 +45,169 @@ def check(*cmd):
...
+        )
+
+    def add_service_account(self, path):
+        """Sets GOOGLE_APPLICATION_CREDENTIALS to path."""
+        self.add_environment('GOOGLE_APPLICATION_CREDENTIALS=%s' % path)
+
+    @property
+    def runner(self):
+        """Finds the best version of e2e-runner.sh."""
+        options = ['e2e-runner.sh', test_infra('jenkins/e2e-image/e2e-runner.sh')]
```
Comment (krzyzacy) [inline scenarios/kubernetes_e2e.py]: It needs to be `/workspace/e2e-runner.sh` when running inside e2e image right?

Comment (fejta) [inline scenarios/kubernetes_e2e.py]: I'm depending on the WORKDIR line: https://github.com/kubernetes/test-infra/blob/master/jenkins/e2e-image/Dockerfile#L68

File: scenarios/kubernetes_e2e.py [hunk 3]
```diff
@@ -96,48 +217,50 @@ def main(args):
     # So if you overwrite FOO=c for a local run it will take precedence.
     #
 
+    # dockerized-e2e-runner goodies setup
+    workspace = os.environ.get('WORKSPACE', os.getcwd())
+    artifacts = '%s/_artifacts' % workspace
+    if not os.path.isdir(artifacts):
+        os.makedirs(artifacts)
+
```
Comment (krzyzacy) [inline scenarios/kubernetes_e2e.py]: nit: one less empty line

File: scenarios/kubernetes_e2e.py [hunk 4]
```diff
@@ -44,49 +45,169 @@ def check(*cmd):
...
+            '-v', '/etc/localtime:/etc/localtime:ro',
+        ]
+        for path in mount_paths:
+            self.cmd.extend(['-v', path])
+
+        if sudo:
+            self.cmd.extend(['-v', '/var/run/docker.sock:/var/run/docker.sock'])
+        self.add_environment(
+            'HOME=/workspace',
+            'WORKSPACE=/workspace')
```
Comment (krzyzacy) [inline scenarios/kubernetes_e2e.py]: wonder if those two would affect local runs, if triggered from prow.

File: scenarios/kubernetes_e2e.py (hunk 3 above)
Comment (fejta) [inline scenarios/kubernetes_e2e.py]: Done

File: scenarios/kubernetes_e2e.py (hunk 4 above)
Comment (fejta) [inline scenarios/kubernetes_e2e.py]: Yeah, I suspect this file will require additional futzing to work on prow.

In this particular instance the image should already set WORKSPACE=/workspace, so it shouldn't matter.

**Comment by fejta** [comment]
```
I0223 18:11:26.422] Starting ci-kubernetes-e2e-gce-canary-20170224-015655-1e80eef7b01f7529-17896...
I0223 18:11:26.423] process 17929 exited with code 0 after 14.5m
I0223 18:11:26.424] PASS: ci-kubernetes-e2e-gce-canary
```

**Comment by fejta** [comment]
Things look healthy: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gke-updown/9322 (note the fact that -e HOME=/workspace now is earlier in the docker run flag list)

File: scenarios/kubernetes_e2e.py (hunk 2 above)
Comment (krzyzacy) [inline scenarios/kubernetes_e2e.py]: http://prow.k8s.io/log?pod=ci-kubernetes-e2e-prow-canary-9 hummmm

**Pull request merged**